ang/volunteer/VolOppsCtrl.js

```diff
@@ -31,6 +31,29 @@
     // on page load, search based on the URL params
     volOppSearch.search();
 
+    //VOL-190: Allow hiding of search pane based on url param
+    if ($route.current.params.hasOwnProperty('hideSearch')) {
```

Comment:
What if we simplified this a bit?

```javascript
$scope.hideSearch = false;
$scope.allowShowSearch = true;

if ($route.current.params.hasOwnProperty('hideSearch')) {
  if ($route.current.params.hideSearch === "1" ||
    $route.current.params.hideSearch === "always"
  ) {
    $scope.hideSearch = true;    
  }
  if ($route.current.params.hideSearch === "always") {
    $scope.allowShowSearch = false;
  }
}
```

This is a slight deviation from your spec, but really we want the default to be that the search form is displayed. To deviate from the default, you should be able to get the value right: '1' or 'always.' No need to be so generous to lazy developers, I say.